interfaces/builtin/custom_device.go

```diff
@@ -451,8 +462,33 @@ func (iface *customDeviceInterface) UDevConnectedPlug(spec *udev.Specification,
 	}
 
 	// Now write all the rules
-	for _, rule := range deviceRules {
-		spec.TagDevice(rule)
+	for deviceName, rule := range deviceRules {
+		if rule == placeholderRule {
+			baseName := filepath.Base(deviceName)
+
+			// There is an explicit rule for the basename of the
+			// device path, which is distinct from the full device
+			// path, but which doesn't exist as its own self-
+			// contained device path of the form /dev/<basename>,
+			// so don't emit a default rule for the full path.
+			// XXX: Is this safe? Is it okay to have multiple
+			// devices with the same basename?
+			if baseNameRule, exists := deviceRules[baseName]; deviceName != baseName && exists && baseNameRule != placeholderRule && !strutil.ListContains(allDevicePaths, "/dev/"+baseName) {
+				continue
```

Comment:
I feel like we might want to log something here or maybe we should log something in the loops above if hit a situation like:

/dev/foo
/dev/nested/foo

?